dist/mozilla/Hacl_Streaming_SHA3.c

```diff
@@ -129,9 +128,10 @@ Hacl_Streaming_SHA3_update_256(
     }
     if (!(sz1 == (uint32_t)0U))
     {
+      uint32_t nb = (uint32_t)1U;
+      for (uint32_t i = (uint32_t)0U; i < nb; i++)
```

Comment:
Missing ` [@inline_let]` for `nb`?

Reply:
weird I don't know where this is coming from

Reply:
`Hacl.Streaming.SHA3` is now using `Lib.Buffer.loop_blocks`, so we just need to mark more things there with `[@inline_let]`: https://github.com/hacl-star/hacl-star/blob/main/lib/Lib.Buffer.fst#L344

Reply:
ok if I merge this and then submit a followup PR? 

just to be clear, you are suggesting marking `nb` in Lib.Buffer as `[@inline_let]`, correct?

Reply:
> ok if I merge this and then submit a followup PR?

yep

> just to be clear, you are suggesting marking nb in Lib.Buffer as [@inline_let], correct?

yes, the code looks a bit better but still contains `n * 136 / 136`

Reply:
yeah I don't think it's generally true that `(m * n)/n` can be rewritten to `m`, right? (in the presence of potential overflows)

so I don't think krml will rewrite that for you... if we care enough, I guess the solution would be to have `loop_blocks_n` that takes a number of blocks instead of a length

Reply:
yes, it's not true in that case. we can just pass `nb` as an extra argument to `loop_blocks` w/o creating a new function (as it's only used for `sha3`).

Reply:
ok I'm taking a stab at it see if it improves the code generation... it's not a big change

Reply:
I fixed it up on this branch, let me know if that looks good and then I can merge? thanks!